reminders/class_models/notification_models.py

```diff
@@ -0,0 +1,63 @@
+from events.models import EventParticipant
+from django.utils import timezone
+from django.db import models
+from django.db.models import Q
+from django.db.models.functions import Now
+from django.core.exceptions import ValidationError
+from django.db import IntegrityError
+
+
+class Notification(models.Model):
+    participant_id = models.ForeignKey(EventParticipant, on_delete=models.CASCADE)
+    seen_time = models.DateTimeField(default=timezone.now, null=True)
+    sent_time = models.DateTimeField(default=timezone.now)
+    message = models.TextField(null=True, blank=True)
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(fields=['participant_id', 'sent_time'], name='unique notification'),
+            models.CheckConstraint(check=Q(sent_time__gte=Now()), name="sent_time__gte_current_time")
```

Comment:
Can you explain please what is Q? and why do we need it?

Reply:
The Q class is to make query inputs with functions inside,
Like in this case we used the Q class to check the seen time against the current time,
but the current time is a "flowing object" so every time we would check it would be a different value
so we used the Now function that gets the current date time to check if the seen time is greater than the current time.

for more examples and further reading:
https://docs.djangoproject.com/en/4.0/topics/db/queries/#complex-lookups-with-q-objects